enshrine-esdt-safe/src/from_sovereign/transfer_tokens.rs

```diff
@@ -208,6 +340,18 @@ pub trait TransferTokensModule:
             .sync_call();
     }
 
+    fn burn_sovereign_tokens(&self, operation: &Operation<Self::Api>) {
+        for token in operation.tokens.iter() {
+            if self.has_sov_prefix(&token.token_identifier, self.get_sovereign_prefix()) {
```

Comment:
To be consistent with mint function, you can use
`if !self.has_prefix { continue }` otherwise burn